core/modules/search/search.admin.inc

```diff
@@ -5,26 +5,60 @@
  */
 
 /**
- * Menu callback: confirm wiping of the index.
+ * Menu callback: confirm rebuilding of the index.
  */
 function search_reindex_confirm() {
-  return confirm_form(array(),
-    t('Are you sure you want to invalidate the search index?'),
-    'admin/config/search/settings',
-    t('The search index is not cleared immediately but will be systematically updated. Search will continue to work but new content won\'t be indexed until all existing content has been re-indexed. This action cannot be undone.'),
-    t('Invalidate search index'),
-    t('Cancel'));
+  $form = confirm_form(array(),
+  t('Are you sure you want to rebuild the search index?'),
+  'admin/config/search/settings',
+  t('This action will rebuild the whole search index. It may be a lengthy process.'),
+  t('Rebuild search index'),
+  t('Cancel'));
+
+  // Change the css class to something less scary than button-danger.
+  $form['actions']['submit']['#attributes']['class'][0] = 'button-primary';
+  return $form;
 }
 
 /**
- * Handler for wipe confirmation
+ * Handler for rebuild confirmation
  */
 function search_reindex_confirm_submit(&$form, &$form_state) {
   if ($form['confirm']) {
-    search_reindex();
-    backdrop_set_message(t('The search index will be invalidated.'));
+    $config = config('search.settings');
+    $limit_per_pass = $config->get('search_cron_limit');  
+
+    // Wipe out all indexes.
+    module_invoke_all('search_reset');
```

Comment:
This line accomplishes the same thing and it might be a good idea to re-use the existing API rather than duplicating calls to module_invoke_all():

```suggestion
    // Mark all content as needing to be re-indexed.
    search_reindex();
```

Reply:
Done.